src/borg/archiver/debug_cmd.py

```diff
@@ -265,6 +265,71 @@ def do_debug_id_hash(self, args, repository, manifest):
         print(id.hex())
         return EXIT_SUCCESS
 
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_parse_obj(self, args, repository, manifest):
+        """parse object file into meta dict and data (decrypting, decompressing)"""
+
+        # get the object from id
+        hex_id = args.id
+        try:
+            id = unhexlify(hex_id)
+            if len(id) != 32:  # 256bit
+                raise ValueError("id must be 256bits or 64 hex digits")
+        except ValueError as err:
+            print("object id %s is invalid [%s]." % (hex_id, str(err)))
+            return EXIT_ERROR
+        try:
+            cdata = repository.get(id)
+        except Repository.ObjectNotFound:
+            print("object %s not found." % hex_id)
+            return EXIT_ERROR
+
+        output_metadata_path: str = args.json_path
+        output_data_path: str = args.data_path
+
+        key = manifest.key
+        repo_objs = RepoObj(key)
+
+        try:
+            meta, data = repo_objs.parse(obj_id=id, cdata=cdata, decompress=True)
+        except Exception as e:
+            print(f"encountered exception {e.__str__}")
+            return EXIT_ERROR
+
+        with open(output_metadata_path, "wb") as f:
+            f.write(json.dump(meta))
+
+        with open(output_data_path, "wb") as f:
+            f.write(data)
+
+        return EXIT_SUCCESS
+
+    @with_repository(compatibility=Manifest.NO_OPERATION_CHECK)
+    def do_debug_format_obj(self, args, repository, manifest):
+        """format file and metadata into borg object file"""
+        with open(args.data_path, "rb") as f:
+            data = f.read()
+
+        with open(args.json_path, "rb") as f:
+            meta = json.load(f)
+
+        key = manifest.key
+        repo_objs = RepoObj(key)
+
+        id = repo_objs.id_hash(data)
```

Comment:
also, this is duplicated from `borg debug id-hash`.

Reply:
In the case of `id-hash` function, instead of returning the value, we are printing the output. Hence we need to may need to rewrite the function or return multiple values.

Reply:
i did not mean that you call that function from here.

what i meant is that the user uses `borg debug id-hash` to get the id.
then the user would invoke `borg debug format-obj ID JSONFILE DATAFILE OBJFILE` (or so) to create a raw obj file (which then could be written to the repo using `borg debug put-obj ...`).